data-serving/data-service/src/model/case.ts

```diff
@@ -114,91 +105,126 @@ const pathogenSchema = new mongoose.Schema({
-
-    country: {
-        type: String,
-        required: 'Enter a country',
-    },
+    country: String,
     administrativeRegion1: String,
     administrativeRegion2: String,
-
     locality: String,
-    geometry: geometrySchema,
-});
-
-const dateRangeSchema = new mongoose.Schema({
-    start: Date,
-    end: Date,
+    geometry: {
+        latitude: {
+            type: Number,
+            min: -90.0,
+            max: 90.0,
+        },
+        longitude: {
+            type: Number,
+            min: -180.0,
+            max: 180.0,
+        },
+    },
 });
 
 const travelSchema = new mongoose.Schema({
-    location: locationSchema,
+    location: {
+        type: locationSchema,
+        validate: minimumLocationFieldsValidator,
+    },
     dates: dateRangeSchema,
     purpose: {
         type: String,
-        enum: Object.keys(TravelPurpose),
+        enum: Object.values(TravelPurpose),
     },
     additionalInformation: String,
 });
 
-const travelHistorySchema = new mongoose.Schema({
-    fields: [travelSchema],
-});
-
-const caseSchema = new mongoose.Schema({
-    chronicDisease: dictionaryValueSchema,
-    demographics: demographicsSchema,
-    eventSequence: {
-        type: eventSequenceSchema,
-        required: 'Enter an event sequence',
+const caseSchema = new mongoose.Schema(
+    {
+        chronicDisease: dictionaryValueSchema,
+        demographics: demographicsSchema,
+        events: {
+            type: [
+                {
+                    name: {
+                        type: String,
+                        required: 'Enter a name for the event',
+                    },
+                    date: dateRangeSchema,
+                },
+            ],
+            validate: {
+                validator: function (events: [Event]) {
+                    return events.some((e) => e.name == 'confirmed');
+                },
+                message: 'Must include a case confirmed event',
+            },
+        },
+        importedCase: {},
+        location: {
+            type: locationSchema,
+            required: 'Enter a location',
+            validate: minimumLocationFieldsValidator,
+        },
+        revisionMetadata: {
+            type: revisionMetadataSchema,
+            required: 'Enter revision metadata',
+        },
+        notes: String,
+        outbreakSpecifics: outbreakSpecificsSchema,
+        pathogens: [pathogenSchema],
+        source: {
+            type: sourceSchema,
+            required: 'Enter a source',
+            validate: minimumSourceFieldsValidator,
+        },
+        symptoms: dictionaryValueSchema,
+        travelHistory: [travelSchema],
     },
-    importedCase: Object,
-    location: locationSchema,
-    metadata: metadataSchema,
-    notes: String,
-    outbreakSpecifics: outbreakSpecificsSchema,
-    outcome: {
-        type: String,
-        enum: Object.keys(Outcome),
-        required: 'Enter an outcome.',
+    {
+        useNestedStrict: true,
```

Comment:
We don't have any nested `strict` settings.

Reply:
hmm I guess I read the documentation wrong -- since it defaults to strict, even if this is 'false', everything will be treated as strict (which is what we want). thanks!